### ISSUE TITLE
Rename chunk & shard caches to L1 & L2 caches

### ISSUE THREAD

**Issue by Longarithm** [near/nearcore#9054]
Long-standing readability issue for someone who wants to do massive renaming :)

We initially named our caches for state data in `TrieCachingStorage` as `shard_cache` (also known as `TrieCache`) and `chunk_cache`. However, this naming is considered poor because these caches are mostly used in runtime, runtime is logically separated from chain, and shards & chunks are chain concepts.

Instead, L1Cache and L2Cache seem better. First one is a nondeterministic LRU cache storing all "trie values" by their hashes among all processed blocks. Second is deterministic but stores updates only for single chunk.

cc @jbajic

**Comment by robin-near** [comment]
I agree on the need to rename the caches.

But maybe we should pick something more explicit than L1 and L2. Because it's already not clear which one is L1 and which one is L2. I thought L1 was the deterministic one and L2 is the LRU... :)

Maybe AccountingCache for the deterministic one and just... LruCache for the LRU one?

**Comment by Longarithm** [comment]
No, you are right. L1 is deterministic and L2 is LRU.

I can't choose... "accounting" also looks too long. Can I ask for more opinions:
* @jakmeier - I remember you suggested L1&L2
* @jbajic - you may have more experience with naming in storage logic :)

**Comment by jakmeier** [comment]
Yeah I suggested `L1TrieNodeCache` and `L2TrieNodeCache`. This make it clear that we first look in L1 and if that fails we go to L2. And `TrieNodeCache` tells us what's stored inside.
This is the exact naming scheme I have seen in my previous job where the code base had several dozen different cache classes in the same code base. The only problem I can remember is that sometime you add a new cache that either between existing caches, which means you might have to rename them in the future and update documentation.

Another option would be to call the shard cache `TrieNodeCache` and for the chunk cache we use `DeterministicTrieNodeCache`. 

Even better IMO would be to change the chunk cache implementation from a cache to just a hash set. Then we could call it something like `CheapTrieNodes` or `RecentlySeenTrieNodeSet`. And the shard cache would become the only `TrieNodeCache` with no extra pre- o or post fixes needed

**Comment by jbajic** [comment]
I think @jakmeier naming convention makes sense to me, just L1 or L2 leaves a lot of context out for the reader, but making it `L1TrieNodeCache` makes it more readable for me, even though it exposes the tree structure that we are using as a implementations detail, I don't mind that.
